src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,119 @@
+import { base64EncodeTokenDetails } from "util/helpers";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+  const [error, setError] = useState(false);
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoading(true);
+
+      createFineGrainedTlsIdentity(values.clientName)
+        .then((tokenPreResponse) => {
+          const tempToken = base64EncodeTokenDetails(tokenPreResponse);
+
+          setIsLoading(false);
+          setToken(tempToken);
+
+          void queryClient.invalidateQueries({
+            queryKey: [queryKeys.identities],
+          });
+        })
+        .catch(() => {
+          setIsLoading(false);
+          setError(true);
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={onClose}
+      className="create-tls-identity"
+      title="Generate trust token"
+      buttonRow={
+        <>
+          {token && (
+            <Button
+              className="u-no-margin--bottom"
+              onClick={() => {
+                navigator.clipboard.writeText(token);
+              }}
+              type="button"
+              hasIcon
+            >
+              <Icon name="copy" />
+            </Button>
+          )}
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            onClick={token ? onClose : () => void formik.submitForm()}
+            disabled={formik.values.clientName.length === 0 || isLoading}
+            loading={isLoading}
+          >
+            {token ? `Close` : `Generate Token`}
+          </ActionButton>
+        </>
+      }
+    >
+      {error && (
+        <Notification
+          severity="negative"
+          title="Fine grained identity token creation failed"
+        ></Notification>
+      )}
+      {!token && (
+        <Input
+          id="clientName"
+          type="text"
+          label="Client Name"
+          onBlur={formik.handleBlur}
+          onChange={formik.handleChange}
+          value={formik.values.clientName}
+        />
+      )}
+      {token && (
+        <>
+          <p>Identity has been created.</p>
+          <p>The token below can be used to log in as the new identity.</p>
+
+          <Notification
+            severity="caution"
+            title="Make sure to copy the token now as it will not be shown again."
+          ></Notification>
+
+          <div className="token-code-block">
+            <code>{token}</code>
+          </div>
```

Comment:
We should reuse the [CodeSnippet](https://canonical.github.io/react-components/?path=/story/components-codesnippet--wrap-lines) and avoid a custom styling for it. The custom styling creates more code and is missing margins.

Reply:
CodeSnippet does not currently support strings without breaks. I have raised this in the Vanilla channel. There are some workarounds however these also require custom styling. Please see (this thread)[https://chat.canonical.com/canonical/pl/44y5kzsagfnsmfyja8hi5wwt6e] and let me know your thoughts.

Reply:
I see, it doesn't wrap long strings. Then, let's add the default margin below the element to fix the spacing.

Reply:
![image](https://github.com/user-attachments/assets/d850bdf4-cf96-4461-bf97-0bb16003df9c)

Something like this, but not as hacky, better with a class and applying the styling via scss.